statline_bq/utils.py

```diff
@@ -470,184 +529,45 @@ def get_column_descriptions_v3(url_data_properties: str) -> dict:
-
-    # check if more data exists
-    target_url = r.get("odata.nextLink", None)
-
-    # if more data exists continue to concat bag until complete
-    while target_url:
-        r = requests.get(target_url).json()
-        if r["value"]:
-            temp_bag = db.from_sequence(r["value"])
-            bag = db.concat([bag, temp_bag])
-
-        target_url = r.get("odata.nextLink", None)
-
-    return bag
-
-
-def get_odata_v4(target_url: str,) -> db:
-    """Gets a table from a specific url for CBS Odata v4.
-    This function uses standard requests.get() to retrieve data at target_url
-    in json format, and concats it all into a Dask Bag to handle memory
-    overflow if needed.
+def get_odata(table_urls: list) -> db:
+    """Create a dask.bag object holding all values of a table for a CBS dataset
 
-    Each request from CBS is limited to 10,000 rows, and if more data exists
-    the key "@odata.nextLink" exists in the response with the link to the next
-    10,000 (or less) rows.
+    This functions maps `load_from_url()` on a list of urls, all related to a single
+    CBS table, from a specific dataset. Since requests to CBS are limited at 10,000
+    rows (=observations/records), the table_urls consists of the same base url, each ending with "$?SKIP={i}",
+    where i is an integer with multiples of 10,000, i.e.:
 
-    Meant to be wrapped by `get_odata()`, and it is better practice to call it
-    wrapped to allow for both "v3" and "v4" functionality.
+    [
+        "https://opendata.cbs.nl/ODataFeed/odata/83583NED/TypedDataSet?$format=json",
+        "https://opendata.cbs.nl/ODataFeed/odata/83583NED/TypedDataSet?$format=json&$skip=10000"
+        "https://opendata.cbs.nl/ODataFeed/odata/83583NED/TypedDataSet?$format=json&$skip=20000"
+        ...
+        ...
+    ]
 
     Parameters
     ----------
-    target_url: str
-        A valid url of a table from CBS
+    table_urls: list of str
+        A list of valid urls of a table from CBS
 
     Returns
     -------
     bag: Dask Bag
-        All data received from target url as json type, concatenated as a Dask bag
+        The values included in table_urls as json type, as a single Dask bag
     """
-
-    print(f"Fetching from {target_url}")
-    # First call target url and get json formatted response as dict
-    r = requests.get(target_url).json()
-    # Create Dask bag from dict
-    bag = db.from_sequence(r["value"])  # TODO -> define npartitions?
-
-    # check if more data exists
-    target_url = r.get("@odata.nextLink", None)
-
-    # if more data exists continue to concat bag until complete
-    while target_url:
-        r = requests.get(target_url).json()
-        temp_bag = db.from_sequence(r["value"])
-        bag = db.concat([bag, temp_bag])
-
-        target_url = r.get("@odata.nextLink", None)
-
+    print("Creating bag")
+    bag = db.from_sequence(table_urls).map(load_from_url)
+    print("Created bag")
     return bag
 
 
 def convert_table_to_parquet(
```

Comment:
@galamit86 

I think this can be done more efficiently by using `bag.to_dataframe().to_parquet`: it converts the Dask object into a Dask dataframe and then to parquet. See https://docs.dask.org/en/latest/dataframe-create.html
That way you don't need to temporary json files. Or am I missing something?

Reply:
It is possible to do, but I will create a separate issue for that - it is certainly more readable, replacing ~20 code lines with 1, but I am not 100% sure it actually is more efficient time-wise, that should be checked. Considering the unnecessary overhead in the dataframe creation, I can imagine we might actually be speeding things up here. Not sure. And since it creates a slightly different output than the current implementation (folders instead of files, mostly) - I will leave it for now.